app/routers/api/browser.py

```
@@ -50,6 +50,37 @@ def list_directory(
+                    # Check if resolved_path is base or a subdirectory of base
+                    # Note: For Python 3.9+, use resolved_path.is_relative_to(base)
+                    # For compatibility, we use relative_to with try/except
+                    resolved_path.relative_to(base)
+                    is_allowed = True
+                    break
+                except ValueError:
+                    continue
+
+            if not is_allowed:
+                logger.warning(
+                    f"Unauthorized directory browse attempt by {current_user.username}: {path}"
```

Ruff (0.14.14)

[warning] Logging statement uses f-string

(G004)